scripts/leapp_upgrade.py

```diff
@@ -0,0 +1,430 @@
+        report_entries = report_json.get("entries", [])
+        inhibitor_count = len(
+            [entry for entry in report_entries if "inhibitor" in entry.get("flags")]
+        )
+        message = "Your system has %s inhibitors out of %s potential problems." % (
+            inhibitor_count,
+            len(report_entries),
+        )
+        if leapp_upgrade_output and REBOOT_GUIDANCE_MESSAGE in leapp_upgrade_output:
+            message += " System is ready to be upgraded. Rebooting system in 1 minute."
+        alert = inhibitor_count > 0
+        status = (
+            _find_highest_report_level(report_entries)
+            if len(report_entries) > 0
+            else "SUCCESS"
+        )
+
+    output.status = status
+    output.report_json = report_json
+    output.alert = alert
+    output.message = message
+
+    print("Reading TXT report")
+    report_txt = "Not found"
+    if os.path.exists(TXT_REPORT_PATH):
+        with open(JSON_REPORT_PATH, mode="r") as handler:
+            report_txt = handler.read()
+
+    output.report = report_txt
+
+
+def update_insights_inventory():
+    """Call insights-client to update insights inventory."""
+    print("Updating system status in Red Hat Insights.")
+    output, returncode = run_subprocess(["/usr/bin/insights-client"])
+
+    if returncode:
+        raise ProcessError(
+            message="Failed to update Insights Inventory by registering the system again. See output the following output: %s"
+            % output,
+            report="insights-client execution exited with code '%s'." % returncode,
+        )
+
+    print("System registered with insights-client successfully.")
+
+
+def reboot_system():
+    print("Rebooting system in 1 minute.")
+    run_subprocess(["/usr/sbin/shutdown", "-r", "1"], wait=False)
+
+
+def main():
+    try:
+        # Exit if not RHEL 7.9 or 8.4
+        dist, version = get_rhel_version()
+        if dist != "rhel" or is_non_eligible_releases(version):
+            raise ProcessError(
+                message="In-place upgrades are supported only on RHEL distributions.",
+                report='Exiting because distribution="%s" and version="%s"'
+                % (dist, version),
+            )
+
+        output = OutputCollector()
+        upgrade_command = ["/usr/bin/leapp", "upgrade", "--report-schema=1.1.0"]
+        rhui_pkgs = setup_leapp(version)
+
+        # Check for RHUI PKGs
+        use_no_rhsm = should_use_no_rhsm_check(len(rhui_pkgs) > 1, upgrade_command)
+        if use_no_rhsm:
+            install_leapp_pkg_corresponding_to_installed_rhui(rhui_pkgs)
+
+        remove_previous_reports()
+        leapp_upgrade_output = execute_upgrade(upgrade_command)
+        parse_results(output, leapp_upgrade_output)
+        update_insights_inventory()
+        print("Leapp upgrade command successfully executed.")
+        if leapp_upgrade_output and REBOOT_GUIDANCE_MESSAGE in leapp_upgrade_output:
```

Comment:
I am just thinking since this condition is there twice, can't we rename it to something like `reboot_required`? then you can pass that bool to parse_results :)